lib/rules/no-useless-escape.js

```diff
@@ -23,6 +23,12 @@ const VALID_STRING_ESCAPES = [
     "\r"
 ];
 
+const VALID_TEMPLATE_ESCAPES = VALID_STRING_ESCAPES.concat([
+    "$",
+    "{",
```

Comment:
That's correct - now that you mention it, we could fine tune this by only allowing these if an expression is found in the template literal